App/automotive/src/main/java/com/amos/infotaimos/viewmodel/NavigationPageViewModel.kt

```diff
@@ -1,28 +1,44 @@
 package com.amos.infotaimos.viewmodel
 
-
 import android.content.ContentResolver
 import android.content.Context
 import android.media.AudioAttributes
 import android.media.MediaPlayer
 import android.net.Uri
 import androidx.lifecycle.ViewModel
 import com.amos.infotaimos.R
-
 import com.amos.infotaimos.model.CarInstanceManager
 import com.amos.infotaimos.model.NavigationService
 
 class NavigationPageViewModel : ViewModel() {
 
-    fun startNavigation(context: Context, delay: Long) {
-        NavigationService.startNavigation(CarInstanceManager.getCarInstance(context), delay)
-    }
-
-    fun stopNavigation(context: Context, delay: Long) {
-        NavigationService.stopNavigation(CarInstanceManager.getCarInstance(context), delay)
+    fun performNavigationAction(context: Context, delay: Long) {
+        val car = CarInstanceManager.getCarInstance(context)
+        NavigationService.apply {
+            if (startTask == null && stopTask == null) {
+                if (navIndicatorLiveData.value == true) {
```

Comment:
hier wirds ja auf einmal sehr defensiv :D ich würde die ganzen checks ob Navigation gestartet wird/ist in eine Funktion innerhalb des Navigationservices packen